File: Entrada de Dados/Alfabetico/alfabetico.scala
import scala.io.StdIn.readLine

object Main {
  def main(args: Array[String]): Unit = {

    var TEXTO: String = ""
    var VALIDACAO: Boolean = false

    while (true) {
      print("Entre uma cadeia alfabética: ")
      TEXTO = readLine()
      VALIDACAO = true

      for (I <- 0 until TEXTO.length) {
        val CARACTERE = TEXTO.charAt(I)
        if (!(CARACTERE.isLetter || CARACTERE.isWhitespace)) {
          VALIDACAO = false
        }
      }

      if (VALIDACAO == false) {
        println("Entrada inválida. Por favor, tente novamente.")
      } else {
        println(s"Você informou a cadeia: $TEXTO")
        return
      }
    }
  }
}

File: Entrada de Dados/Alfabetico/alfabetico.jl
function validarCadeiaAlfabetica()
  
  TEXTO = ""
  
  while true
    print("Entre uma cadeia alfabética: ")
    TEXTO = readline()
    TEXTO = strip(TEXTO)
    VALIDACAO = false
    for CARACTERE in TEXTO
      if (('A' <= CARACTERE <= 'Z') || 
          ('a' <= CARACTERE <= 'z') || 
                 (CARACTERE == ' '))
        VALIDACAO = true
        break
      end
    end
    if (VALIDACAO == false)
      println("Entrada inválida. Por favor, tente novamente.")
    else
      break
    end
  end
  println("Você informou a cadeia: ", TEXTO)
end

validarCadeiaAlfabetica()
